### PR TITLE
Fix 6529 - Trim overlay name

### LINKED ISSUE #6529: `overlay use 'quoted-module-name'` creates an overlay that cannot be removed
### Describe the bug

if you `overlay use 'test'`, the overlay is created with the name `'test'`, but cannot be removed with `overlay hide` nor `overlay hide 'test'`

### How to reproduce

```nu

> module test {}
> overlay use 'test'
> overlay hide test

Error: nu::shell::overlay_not_found (link)

  × Overlay 'test' not found
   ╭─[entry #5:1:1]
 1 │ overlay hide test
   ·              ──┬─
   ·                ╰── overlay not found
   ╰────

> overlay hide 'test'
Error: nu::parser::active_overlay_not_found (link)

  × Active overlay not found.
   ╭─[entry #6:1:1]
 1 │ overlay hide 'test'
   ·              ───┬──
   ·                 ╰── not an active overlay
   ╰────

> overlay list
╭───┬────────╮
│ 0 │ zero   │
│ 1 │ 'test' │
╰───┴────────╯

```

### Expected behavior

either 
1. Throw an error saying that `'` is an illegal overlay identifier character
2. nu ignores the quotes, and creates the overlay with the name `test`
3. nu creates the overlay with the name `'test'`, and `overlay hide 'test'` succeeds

### Screenshots

image

### Configuration

| key                | value                                           |
| ------------------ | ----------------------------------------------- |
| version            | 0.68.1                                          |
| branch             |                                                 |
| commit_hash        | e76b3d61deef832adce76b0f73f19131de8cbc9e        |
| build_os           | windows-x86_64                                  |
| build_target       | x86_64-pc-windows-msvc                          |
| rust_version       | rustc 1.63.0 (4b91a6ea7 2022-08-08)             |
| rust_channel       | stable-x86_64-pc-windows-msvc                   |
| cargo_version      | cargo 1.63.0 (fd9c4297c 2022-07-01)             |
| pkg_version        | 0.68.1                                          |
| build_time         | 2022-09-08 21:41:56 +00:00                      |
| build_rust_channel | release                                         |
| features           | database, dataframe, default, trash, which, zip |
| installed_plugins  |                                                 |

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Kangaxx-0**
# Description

Fix #6529, the reason why ```overlay hide 'test' ``` does not work is that the signature of its requirement param is a `String`, string value gets trimmed during parsing for all circumstances.

The fix is to trim the quotes in `overlay use`,  which means  ```overlay use  'test'``` and ```overlay use  "test" ```  -> ```overlay use   test ``` 

# Tests

Make sure you've done the following:

- [x] Add tests that cover your changes, either in the command examples, the crate/tests folder, or in the /tests folder.
- [ ] Try to think about corner cases and various ways how your changes could break. Cover them with tests.
- [ ] If adding tests is not possible, please document in the PR body a minimal example with steps on how to reproduce so one can verify your change works.

Make sure you've run and fixed any issues with these commands:

- [x] `cargo fmt --all -- --check` to check standard code formatting (`cargo fmt --all` applies these changes)
- [x] `cargo clippy --workspace --features=extra -- -D warnings -D clippy::unwrap_used -A clippy::needless_collect` to check that you're using the standard code style
- [x] `cargo test --workspace --features=extra` to check that all the tests pass

### CODE REVIEW

**Comment by kubouch** [comment]
Thanks! Can you also test the `overlay use` -> `overlay hide` roundtrip works as well with the quotes?

Also, I'm pretty sure there is a string trimming function somewhere already, you shouldn't need to write your own. It would be good to reuse it to reduce bug surface.

I'm quite surprised the quotes don't get trimmed automatically, I'm wondering why is that...

File: tests/overlays/mod.rs [hunk 1]
```diff
@@ -978,3 +978,29 @@ fn overlay_preserve_hidden_alias() {
     assert_eq!(actual.out, "foo");
     assert_eq!(actual_repl.out, "foo");
 }
+
+#[test]
+fn overlay_trim_signle_quote() {
```
Comment (sholderbach) [inline tests/overlays/mod.rs:983]:
```suggestion
fn overlay_trim_single_quote() {
```

**Comment by Kangaxx-0** [comment]
> Thanks! Can you also test the `overlay use` -> `overlay hide` roundtrip works as well with the quotes?

Sorry, not super clear what `roundtrip` means, can you elaborate a bit please ? :)

> I'm quite surprised the quotes don't get trimmed automatically, I'm wondering why is that...
I believe this is what happened - In `overlay use`, it was a trimmed `String` expr in the first place when internal_call was composed, but how the trimming works is it does not change input spans, `parse_keyword` explicitly read the spans (with quotes) and convert expression to  `Overlay`

**Comment by kubouch** [comment]
Ah, sorry, I meant checking that `overlay use 'spam'` followed by `overlay hide spam` or `overlay hide 'spam'` works.

> I believe this is what happened - In overlay use, it was a trimmed String expr in the first place when internal_call was composed, but how the trimming works is it does not change input spans, parse_keyword explicitly read the spans (with quotes) and convert expression to Overlay.

I think that might be the root cause. I wrote the Overlay expr instead of using String in order to pass the module ID inside but it might not trim the string so maybe it introduced this bug.

Anyway, I think we can merge this fix if you could add the last test with the hiding.

**Comment by Kangaxx-0** [comment]
> Ah, sorry, I meant checking that `overlay use 'spam'` followed by `overlay hide spam` or `overlay hide 'spam'` works.
> 
> > I believe this is what happened - In overlay use, it was a trimmed String expr in the first place when internal_call was composed, but how the trimming works is it does not change input spans, parse_keyword explicitly read the spans (with quotes) and convert expression to Overlay.
> 
> I think that might be the root cause. I wrote the Overlay expr instead of using String in order to pass the module ID inside but it might not trim the string so maybe it introduced this bug.
> 
> Anyway, I think we can merge this fix if you could add the last test with the hiding.

Added two more tests :)

**Comment by kubouch** [comment]
Thanks!

**Pull request merged**